src/libraries/Common/src/Interop/Windows/Kernel32/Interop.DeviceIoControl.cs

```diff
@@ -15,6 +15,44 @@ internal static partial class Kernel32
         // https://docs.microsoft.com/windows-hardware/drivers/ddi/ntddstor/ni-ntddstor-ioctl_storage_read_capacity
         internal const int IOCTL_STORAGE_READ_CAPACITY = 0x002D5140;
 
+        // https://learn.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_set_sparse
+        internal const int FSCTL_SET_SPARSE = 0x000900c4;
+        internal struct FILE_SET_SPARSE_BUFFER
+        {
+            internal int SetSparse;
+        }
+
+        // https://learn.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_get_integrity_information
+        internal const int FSCTL_GET_INTEGRITY_INFORMATION = 0x0009027C;
+        internal struct FSCTL_GET_INTEGRITY_INFORMATION_BUFFER
+        {
+            internal ushort ChecksumAlgorithm;
+            internal ushort Reserved;
+            internal uint Flags;
+            internal uint ChecksumChunkSizeInBytes;
+            internal uint ClusterSizeInBytes;
+        }
+
+        // https://learn.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_set_integrity_information
+        internal const int FSCTL_SET_INTEGRITY_INFORMATION = 0x0009C280;
+        internal struct FSCTL_SET_INTEGRITY_INFORMATION_BUFFER
+        {
+            internal ushort ChecksumAlgorithm;
+            internal ushort Reserved;
+            internal uint Flags;
+        }
+
+        // https://learn.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_duplicate_extents_to_file
+        internal const int FSCTL_DUPLICATE_EXTENTS_TO_FILE = 0x00098344;
+        [StructLayout(LayoutKind.Sequential, Pack = 4)] //the longs are aligned to 4 bytes on 32-bit, and 8 on 64-bit
```

Comment:
Is this really needed? The layout should be satisfied without the `Pack`.

Reply:
Without padding, on 32 bit it would insert 4 bytes of padding before the long, wouldn't it? Or are longs only aligned to 4 bytes on 32 bit?

Reply:
My understanding is that `LARGE_INTEGER` is aligned to 4 bytes - at least on 32 bit, idk about on 64 bit.

Reply:
You can test it, in both managed and native code. Or you can ask CsWin32 and see if it can give any advice (but don't trust it unconditionally).

Reply:
> My understanding is that LARGE_INTEGER is aligned/packed to 4 bytes

That is not correct. LARGE_INTEGER layout is aligned to 8 bytes on both 32-bit and 64-bit.

Reply:
Thanks @jkotas, you are right. I will fix it.